backend/community/guides/views.py

```diff
@@ -0,0 +1,34 @@
+from django.db import transaction
+from django.db.models import Q
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+
+from guides.filters import GuideFilter
+from guides.models import Guide
+from guides.permissions import IsOwnerOrPublic
+from guides.serializers import GuideSerializer, GuideUploadSerializer
+
+
+class GuideViewSet(viewsets.ModelViewSet):
+    queryset = Guide.objects.all().order_by('-published_at')
+    serializer_class = GuideSerializer
+    permission_classes = [IsAuthenticatedOrReadOnly, IsOwnerOrPublic]
+    filter_backends = [DjangoFilterBackend]
+    filterset_class = GuideFilter
+
+    def get_queryset(self):
+        if self.action == 'list':
+            user = self.request.user
+            return super(GuideViewSet, self).get_queryset().filter(Q(private=False) | Q(owner=user.id))
+        else:
+            return super(GuideViewSet, self).get_queryset()
+
+    def get_serializer_class(self):
+        if self.action == 'create':
+            return GuideUploadSerializer
+        return GuideSerializer
+
+    @transaction.atomic
```

Comment:
Is it not enough to have `@transaction.atomic` only on serializer's `create` (as I've seen it is there)?

Reply:
The transaction per se is not required as we only have a single sql operation here .
I only use it to have the callback when the object is actually written to the database to avoid race-condition with the scheduled selery task.